inotify/src/lib.rs

```diff
@@ -264,10 +265,17 @@ impl Inotify {
     /// [`Inotify::add_watch`]: struct.Inotify.html#method.add_watch
     /// [`Event`]: struct.Event.html
     pub fn rm_watch(&mut self, wd: WatchDescriptor) -> io::Result<()> {
-        let result = unsafe { ffi::inotify_rm_watch(self.0, wd.0) };
+        let result: c_int;
+        if self.0 == wd.1 {
+            result = unsafe { ffi::inotify_rm_watch(self.0, wd.0) }; 
+        }
+        else {
+            result = -2;
```

Comment:
Thanks for the feedback!

Yes, you're right, it does make sense to not pollute `result`. I'll fix this soon.